libs/hdf-converters/src/ckl-mapper/checklist-mapper.ts

```diff
@@ -230,6 +231,31 @@ export function getChecklistObjectFromHdf(
   return _.get(hdf, 'passthrough.checklist', EmptyChecklistObject);
 }
 
+export function validateChecklistMetadata(
+  metadata: Pick<Asset, 'hostip' | 'hostfqdn' | 'hostmac'>
+): {isError: boolean; message: string; invalid: string[]} {
```

Comment:
Let's explore using an actual Results type monoid as opposed to this unless using that type turns out to be a pain for some reason

Reply:
I created a simple Result<T, E> type for this purpose.